app/services/cd_api/search_service.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module CdApi
+  class SearchService
+    def self.call(filter:, term:, dob:)
+      new(filter, term, dob).call
+    end
+
+    def initialize(filter, term, dob)
+      @filter = filter
+      @term = term
+      @dob = dob.to_s if dob # TODO: request the api to use date object instead of string or be consistent with v1
+    end
+
+    def call
```

Comment:
Do we need to have a switch statement here? The term doesnt really matter as all params go into the same call - we just need a param builder which should reduce the cognitive complexity and maintenance here.